wallet/src/de/schildbach/wallet/livedata/DecryptSeedLiveData.kt

```diff
@@ -1,88 +0,0 @@
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package de.schildbach.wallet.livedata
-
-import android.os.Handler
-import android.os.HandlerThread
-import android.os.Process
-import androidx.lifecycle.MutableLiveData
-import de.schildbach.wallet.security.SecurityGuard
-import de.schildbach.wallet.payments.DecryptSeedTask
-import de.schildbach.wallet.payments.DeriveKeyTask
-import org.bitcoinj.wallet.DeterministicSeed
-import org.bitcoinj.wallet.Wallet
-import org.bouncycastle.crypto.params.KeyParameter
-
-/**
- * @author:  Eric Britten
- */
-
-class DecryptSeedLiveData(
-    private val wallet: Wallet,
-    private val scryptIterationsTarget: Int
-) : MutableLiveData<Resource<Pair<DeterministicSeed?, String?>>>() {
-
-    val backgroundHandler: Handler
-
-    init {
-        val backgroundThread = HandlerThread("backgroundThread", Process.THREAD_PRIORITY_BACKGROUND)
-        backgroundThread.start()
-        backgroundHandler = Handler(backgroundThread.looper)
-    }
-
-    private var decryptSeedTask: DecryptSeedTask? = null
-    private var deriveKeyTask: DeriveKeyTask? = null
-
-    private val securityGuard = SecurityGuard()
-
-    fun checkPin(pin: String) {
-        if (securityGuard.checkPin(pin)) {
-            decryptSeed(pin, securityGuard.retrievePassword())
-        } else {
-            value = Resource.error("wrong pin", Pair(wallet.keyChainSeed, pin))
-        }
-    }
-
-    private fun decryptSeed(pin: String, password: String) {
-        if (deriveKeyTask == null) {
-            deriveKeyTask = object : DeriveKeyTask(backgroundHandler, scryptIterationsTarget) {
-
-                override fun onSuccess(encryptionKey: KeyParameter, changed: Boolean) {
-                    deriveKeyTask = null
-                    if (decryptSeedTask == null) {
-                        decryptSeedTask = object : DecryptSeedTask(backgroundHandler) {
-
-                            override fun onBadPassphrase() {
-                                value = Resource.error("wrong password", Pair(wallet.keyChainSeed, pin))
-                                decryptSeedTask = null
-                            }
-
-                            override fun onSuccess(seed : DeterministicSeed) {
-                                value = Resource.success(Pair(seed, pin))
-                                decryptSeedTask = null
-                            }
-                        }
-                        value = Resource.loading(null)
-                        decryptSeedTask!!.decryptSeed(wallet.keyChainSeed, wallet.keyCrypter, encryptionKey)
-                    }
-                }
-            }
-            value = Resource.loading(null)
-            deriveKeyTask!!.deriveKey(wallet, password)
-        }
```

Comment:
I think it's better to have this logic as a couple of `suspend` calls in the `SecurityFunctions` class. That way, we can use it from anywhere - not just the UI thread.